lib/platform/comment.ts

```diff
@@ -0,0 +1,59 @@
+import hasha from 'hasha';
+import { getCache } from '../util/cache/repository';
+import type {
+  EnsureCommentConfig,
+  EnsureCommentRemovalConfigByContent,
+  EnsureCommentRemovalConfigByTopic,
+} from './types';
+import { platform } from '.';
+
+const hash = (content: string): string => hasha(content, { algorithm: 'sha1' });
+
+export async function ensureComment(
+  commentConfig: EnsureCommentConfig
+): Promise<boolean> {
+  const { number, content } = commentConfig;
+  const topic = commentConfig.topic ?? '';
+
+  const contentHash = hash(content);
+  const repoCache = getCache();
+
+  if (contentHash !== repoCache.prComments?.[number]?.[topic]) {
+    const res = await platform.ensureComment(commentConfig);
+    if (res) {
+      repoCache.prComments ??= {};
+      repoCache.prComments[number] ??= {};
+      repoCache.prComments[number][topic] = contentHash;
+    }
+    return res;
+  }
+
+  return true;
+}
+
+export async function ensureCommentRemoval(
+  config:
+    | EnsureCommentRemovalConfigByTopic
+    | EnsureCommentRemovalConfigByContent
+): Promise<void> {
+  await platform.ensureCommentRemoval(config);
+
+  const { number, topic, content } = { topic: null, content: null, ...config };
+  const repoCache = getCache();
+  if (repoCache.prComments?.[number]) {
+    if (topic) {
```

Comment:
I thought that you saved the content hash to topic `''`, so why check if it's truthy when you're removing?

Reply:
It was related not to topic nullability, to distinguish `by-topic`/`by-content`. Now this field is explicit.